McCorrections/include/TauIdWeight.h

```diff
@@ -131,9 +131,117 @@ class TauIdWeight : public IWeightProvider {
     }
 
     std::map<Key, Parameters> tauIdparam_map;
-    DiscriminatorWP iso_wp;
-
 };
 
+class TauIdWeight2017 : public TauIdWeight {
+public:
+    virtual double GetIdIsoSF(const LorentzVectorM_Float& p4, GenMatch /*gen_match*/, int /*decay_mode*/, DiscriminatorWP anti_ele_wp,
+                             DiscriminatorWP anti_mu_wp, DiscriminatorWP iso_wp) const override
+    {
+        auto tauSF = getTauIso(iso_wp);
+        auto muonSF = getMuonMissIdSF(p4, anti_mu_wp);
+        auto eleSF = getEleMissIdSF(p4, anti_ele_wp);
+
+        return tauSF * muonSF * eleSF;
+    }
+
+private:
+    double getMuonMissIdSF(const LorentzVectorM_Float& p4, DiscriminatorWP iso_wp) const {
+        // https://indico.cern.ch/event/738043/contributions/3048471/attachments/1674773/2691664/TauId_26062018.pdf
+        // https://twiki.cern.ch/twiki/bin/viewauth/CMS/MuonReferenceEffs2017
+
+        if(std::abs(p4.eta()) < 0.4){
+            if(iso_wp == DiscriminatorWP::Loose)
+                return 1.06;
+            else if(iso_wp == DiscriminatorWP::Tight)
+                return 1.17;
+            else throw exception("WP %1% is not supported.") % iso_wp;
```

Comment:
move check of wp in the beginning of the method. then, inside if for eta use:
```return iso_wp == DiscriminatorWP::Loose ? 1.06 : 1.17;``` etc.